ehsm_kms_service/function.js

```diff
@@ -444,71 +381,92 @@ const _checkParams = function (req, res, next, nonce_database, DB) {
       (payload && typeof payload != 'object') ||
       typeof sign != 'string'
     ) {
+      ReportControlTarget(req.ip, new Date().getTime())
       res.send(_result(400, 'param type error'))
       return
     }
     if (timestamp.length != TIMESTAMP_LEN) {
+      ReportControlTarget(req.ip, new Date().getTime())
       res.send(_result(400, 'Timestamp length error'))
       return
     }
     if (!_checkTimestamp(timestamp)) {
+      ReportControlTarget(req.ip, new Date().getTime())
       res.send(_result(400, 'Timestamp error'))
       return
     }
-    /**
-     * Cache nonce locally after receiving the request
-     * nonce_database - object
-     *  {
-     *    <appid>: [nonce_data，nonce_data]
-     *  }
-     * nonce_data - object
-     *  {
-     *    nonce: ****
-     *    nonce_timestamp: new Date().getTime()
-     *  }
-     */
-    const nonce_data = { nonce, nonce_timestamp: new Date().getTime() }
-    if (!nonce_database[appid]) {
-      nonce_database[appid] = [nonce_data]
-    } else if (
-      !!nonce_database[appid] &&
-      nonce_database[appid].findIndex(
-        (nonce_data) => nonce_data.nonce == nonce
-      ) > -1
-    ) {
-      res.send(_result(400, "Timestamp can't be repeated in 20 minutes"))
-      return
-    } else {
-      nonce_database[appid].unshift(nonce_data)
-    }
 
-    if (ACTION !== key_management_apis[ACTION]) {
-      // check payload
-      const _checkPayload_res = _checkPayload(req, res, next)
-      if (!_checkPayload_res) {
-        return
+    var _lastSteps = () => {

```

Comment:
Not understand

Reply:
_checkParamsis a async function, but this function _checkParams is not a async function.
There should wait the _checkParams result but could not use "await".
So I make the last steps of checking params as a function for async call.